commands/scan/scan.go

```diff
@@ -254,50 +252,35 @@ func (scanCmd *ScanCommand) RunAndRecordResults(recordResFunc func(scanResults *
 	if err = errGroup.Wait(); err != nil {
 		err = errors.New("failed while trying to get Analyzer Manager: " + err.Error())
 	}
-	// resultsArr is a two-dimensional array. Each array in it contains a list of ScanResponses that were requested and collected by a specific thread.
-	resultsArr := make([][]*ScanInfo, threads)
-	fileProducerConsumer := parallel.NewRunner(scanCmd.threads, 20000, false)
-	fileProducerErrors := make([][]formats.SimpleJsonError, threads)
-	indexedFileProducerConsumer := parallel.NewRunner(scanCmd.threads, 20000, false)
-	indexedFileProducerErrors := make([][]formats.SimpleJsonError, threads)
+	fileProducerConsumer := parallel.NewRunner(threads, 20000, false)
+	indexedFileProducerConsumer := parallel.NewRunner(threads, 20000, false)
 	fileCollectingErrorsQueue := clientutils.NewErrorsQueue(1)
 	// Parallel security runner for JAS scans
-	JasScanProducerConsumer := utils.NewSecurityParallelRunner(scanCmd.threads)
-	jasScanProducerErrors := make([][]formats.SimpleJsonError, threads)
+	JasScanProducerConsumer := utils.NewSecurityParallelRunner(threads)
+
 	// Start walking on the filesystem to "produce" files that match the given pattern
 	// while the consumer uses the indexer to index those files.
-	scanCmd.prepareScanTasks(fileProducerConsumer, indexedFileProducerConsumer, &JasScanProducerConsumer, scanResults.ExtendedScanResults.EntitledForJas, resultsArr, fileProducerErrors, indexedFileProducerErrors, jasScanProducerErrors, fileCollectingErrorsQueue, xrayVersion)
+	scanCmd.prepareScanTasks(fileProducerConsumer, indexedFileProducerConsumer, &JasScanProducerConsumer, cmdResults, fileCollectingErrorsQueue)
 	scanCmd.performScanTasks(fileProducerConsumer, indexedFileProducerConsumer, &JasScanProducerConsumer)
 
 	// Handle results
-	flatResults := []*utils.ScaScanResult{}
-
-	for _, arr := range resultsArr {
```

Comment:
where this error collecting process has moved to?

Reply:
no need to collect all to a single place. we have it in SecurityCommandResults